### PR TITLE
Private Infra

### PR DESCRIPTION
**Pull request by thpang**

### CODE REVIEW

File: docs/CONFIG-VARS.md [hunk 1]
```diff
@@ -60,21 +60,24 @@ You can use either static credentials or the name of an AWS profile. If both are
...
-To do this, specify ranges of IP addresses in [CIDR notation](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing).
-Contact your Network Administrator to find the public CIDR range of your network.
+By default, the pubic endpoints of the AWS resources that are being created are only accessible through authenticated AWS clients (for example, the AWS Portal, the AWS CLI, etc.).
+To enable access for other administrative client applications (for example `kubectl`, `psql`, etc.), you can set Security Group rules to control access from your source IP addresses.
 
-Please note that these values are used to add Ingress rules into to an AWS Security Group that by default gets created by terraform. If you specify a pre-existing  [Security Group](#use_existing), you need to add your access CIDRs to that Security Group yourself. The values in this section will not be applied.
+To do set these permissions as part of this Terraform script, specify ranges of IP addresses in [CIDR notation](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing). Contact your Network Administrator to find the public CIDR range of your network.
```
Comment (thpang) [inline docs/CONFIG-VARS.md:66]: The verbiage here is off I believe: `To do set` -> `To do this, set`

File: modules/aws_vpc/main.tf [hunk 2]
```diff
@@ -34,25 +34,6 @@ resource "aws_vpc" "vpc" {
...
-      "Name" = format("%s", "${var.name}-private-endpoint-${var.vpc_private_endpoints[count.index]}")
-    },
-    var.tags,
-  )
-
-  subnet_ids = [ 
-    for subnet in local.private_subnets : subnet.id
-  ]
-}
-
```
Comment (thpang) [inline modules/aws_vpc/main.tf:55]: This code was here in order to force the items into using the private_endpoints for everything internally regardless if it was a pubic/private cluster. What/how is this handled now? I see a lot of logic removed or these items below. This code kept the infra locked down to help in both the public/private scenarios.

File: variables.tf [hunk 3]
```diff
@@ -511,23 +511,17 @@ variable "create_static_kubeconfig" {
...
+    error_message = "ERROR: Supported values for `cluster_api_mode` are - public, private."
   }
 }
 
-variable "vpc_private_endpoints" {
-   description = "Endpoints needed for private cluster"
-   type        = list(string)
-   default     = [ "ec2", "ecr.api", "ecr.dkr", "s3", "logs", "sts", "elasticloadbalancing", "autoscaling" ]
-}
-
```
Comment (thpang) [inline variables.tf:530]: All of these items, as noted above, were in place to keep the infra private internally to help with security. Need to know what/how these items are now handled and locked down as stated above.

File: modules/aws_vpc/variables.tf [hunk 4]
```diff
@@ -103,18 +103,6 @@ variable "map_public_ip_on_launch" {
...
-  description = "Use Private IP address for cluster API endpoint"
-  type        = bool
-  default     = false
-}
-
-variable "vpc_private_endpoints" {
-   description = "Endpoints needed for private cluster"
-   type        = list(string)
-   default     = [ "ec2", "ecr.api", "ecr.dkr", "s3", "logs", "sts", "elasticloadbalancing", "autoscaling" ]
-}
```
Comment (thpang) [inline modules/aws_vpc/variables.tf:116]: All of these items, as noted above, were in place to keep the infra private internally to help with security. Need to know what/how these items are now handled and locked down as stated above.

**Review by thpang — COMMENTED** [review]
Clarification and/or comments needed. Thx.

**Comment by thpang** [comment]
@enderm were you able to respond/update code based on comments here?

Testing meeting on Thursday of this week to cover the code here. Will update with findings and then merge once signed off.

**Comment by enderm** [comment]
@thpang yes, pls see https://github.com/sassoftware/viya4-iac-aws/pull/96/commits/1516c385640bb463df9ebe7419df154cea3d1adf

**Comment by thpang** [comment]
Updates look good. Looking to have testing look over the changes then move forward with a merge after that. Thx again.

File: locals.tf [hunk 5]
```diff
@@ -15,11 +15,15 @@ locals {
   cluster_endpoint_private_access_cidrs = var.cluster_endpoint_private_access_cidrs == null ? distinct(concat(module.vpc.public_subnet_cidrs, module.vpc.private_subnet_cidrs)) : var.cluster_endpoint_private_access_cidrs
   postgres_public_access_cidrs          = var.postgres_public_access_cidrs == null ? local.default_public_access_cidrs : var.postgres_public_access_cidrs
 
+  # IPs
+  create_jump_public_ip                 = var.create_jump_public_ip == null ? true : var.create_jump_public_ip
+  create_nfs_public_ip                  = var.create_nfs_public_ip == null ? false : var.create_nfs_public_ip
```
Comment (thpang) [inline locals.tf]: Very confused here. These locals were removed from the Azure and the variable defaults were added to the variables.tf file and  yet they are now added here. Also the GCP code base sets these defaults as well. Need to keep consistency across the providers for our sanity. Please set the default correctly and remove these locals or update the other 2 repos to reflect the need for the locals. The items needed for the variables.tf are listed below.

```terraform
variable "create_nfs_public_ip" {
  default = false
  type = bool
}

variable "create_jump_public_ip" {
  default = true
  type = bool
}
```

File: locals.tf [hunk 6]
```diff
@@ -15,11 +15,15 @@ locals {
...
+
 
   # Subnets
-  jump_vm_subnet                        = var.create_jump_public_ip ? module.vpc.public_subnets[0] : module.vpc.private_subnets[0]
-  nfs_vm_subnet                         = var.create_nfs_public_ip ? module.vpc.public_subnets[0] : module.vpc.private_subnets[0]
-  nfs_vm_subnet_az                      = var.create_nfs_public_ip ? module.vpc.public_subnet_azs[0] : module.vpc.private_subnet_azs[0]
+  jump_vm_subnet                        = local.create_jump_public_ip ? module.vpc.public_subnets[0] : module.vpc.private_subnets[0]
+  nfs_vm_subnet                         = local.create_nfs_public_ip ? module.vpc.public_subnets[0] : module.vpc.private_subnets[0]
+  nfs_vm_subnet_az                      = local.create_nfs_public_ip ? module.vpc.public_subnet_azs[0] : module.vpc.private_subnet_azs[0]
```
Comment (thpang) [inline locals.tf]: Convert back to `var` from `local`

**Review by thpang — COMMENTED** [review]
Changes to be made for consistency.

**Review by riragh — APPROVED** [review]

**Review by jarpat — APPROVED** [review]

**Comment by thpang** [comment]
Not sure the `local` vs `var` issues were resolved. Can someone investigate these comments and changes?

File: vms.tf [hunk 7]
```diff
@@ -122,7 +122,7 @@ module "nfs" {
   tags               = var.tags
   subnet_id          = local.nfs_vm_subnet
   security_group_ids = [local.security_group_id, local.workers_security_group_id]
-  create_public_ip   = var.create_nfs_public_ip
+  create_public_ip   = local.create_nfs_public_ip
```
[the remark this answers is not in the document]
Comment (jarpat) [reply vms.tf]: This was changed back to vars
https://github.com/sassoftware/viya4-iac-aws/pull/96/files#diff-b3a2446ae4c335ba18343a96f6f25ab27f6ee4127bdbac5d5268e70d53f259a1R125

File: security.tf [hunk 8]
```diff
@@ -21,8 +21,8 @@ resource "aws_security_group" "sg" {
 
 resource "aws_security_group_rule" "vms" {
   count                       = ( length(local.vm_public_access_cidrs) > 0
-                                  && (   (var.create_jump_public_ip && var.create_jump_vm )
-                                      || (var.create_nfs_public_ip && var.storage_type == "standard")
+                                  && (   (local.create_jump_public_ip && var.create_jump_vm )
+                                      || (local.create_nfs_public_ip && var.storage_type == "standard")
```
[the remark this answers is not in the document]
Comment (jarpat) [reply security.tf]: This was changed back to var

https://github.com/sassoftware/viya4-iac-aws/pull/96/files#diff-a997863c31aac48cc94efb55341feb0f37e34f67848f736a6a53aa441434321bR24

File: locals.tf (hunk 6 above)
Comment (jarpat) [reply locals.tf]: This was changed back to var

https://github.com/sassoftware/viya4-iac-aws/pull/96/files#diff-9a576c4980b4f226b7ef1042670d564777bbb0ed8ae2d1df33afbd03e2991fcdR19

**Pull request merged**